key_management/src/keystore.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::errors::Error;
+use crypto::SignatureType;
+use std::collections::HashMap;
+
+/// KeyInfo struct, this contains the type of key (stored as a string) and the private key.
+/// note how the private key is stored as a byte vector
+#[derive(Clone, PartialEq, Debug, Eq)]
+pub struct KeyInfo {
+    key_type: SignatureType,
+    // Vec<u8> is used because The private keys for BLS and SECP256K1 are not of the same type
+    private_key: Vec<u8>,
+}
+
+impl KeyInfo {
+    /// Return a new KeyInfo given the key_type and private_key
+    pub fn new(key_type: SignatureType, private_key: Vec<u8>) -> Self {
+        KeyInfo {
+            key_type,
+            private_key,
+        }
+    }
+
+    /// Return a clone of the key_type
+    pub fn key_type(&self) -> &SignatureType {
+        &self.key_type
+    }
+
+    /// Return a clone of the private_key
+    pub fn private_key(&self) -> &Vec<u8> {
+        &self.private_key
+    }
+}
+
+/// KeyStore struct, this contains a HashMap that is a set of KeyInfos resolved by their Address
+pub trait KeyStore {
+    fn list(&self) -> Vec<String>;
+    fn get(&self, k: &str) -> Result<&KeyInfo, Error>;
```

Comment:
```suggestion
    fn get(&self, k: &str) -> Result<KeyInfo, Error>;
```
This trait only works for mem key store because it has the key info in memory, for a persisted storage this won't be the case